facade/service.go

```diff
@@ -1954,24 +1988,69 @@ func (f *Facade) restoreIPs(ctx datastore.Context, svc *service.Service) error {
 	return nil
 }
 
-func (f *Facade) AssignIPs(ctx datastore.Context, request addressassignment.AssignmentRequest) error {
-	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.AssignIPs"))
+func IpVisitorFn(command string, request addressassignment.AssignmentRequest, f *Facade, ctx datastore.Context) service.Visit {
 	visitor := func(svc *service.Service) error {
-		// get all of the ports for the service
-		portmap, err := GetPorts(svc.Endpoints)
-		if err != nil {
-			glog.Errorf("Could not get ports for service %s (%s): %s", svc.Name, svc.ID, err)
-		} else if len(portmap) == 0 {
+		var portmap Ports
+		var auditMessage string
+		if command == "assign-ip" {
```

Comment:
Why are you string matching here?
You should have separate functions based on the actions that you are trying to perform.